pkg/k8s/watchers/cilium_node.go

```diff
@@ -36,9 +36,6 @@ func (k *K8sWatcher) ciliumNodeInit(ciliumNPClient *k8s.K8sCiliumClient, asyncCo
 					if ciliumNode := k8s.ObjToCiliumNode(obj); ciliumNode != nil {
 						valid = true
 						n := nodeTypes.ParseCiliumNode(ciliumNode)
-						if n.IsLocal() {
-							return
-						}
```

Comment:
Ah, this is why it seemed OK to make the same check under `NodeUpdated()`. Except that `NodeUpdated()` is now called for the local node both from here as the k8s watcher as well as the `NodeDiscovery` object. The NodeDiscovery case deliberately did more previously, including populating the health IPs (which are necessary for the CI to pass).

Hopefully this is OK to just go ahead and remove, but I guess we'll find out if not.

Reply:
^^ FYI we reinstated this code.